lua/wire/client/thrusterlib.lua

```diff
@@ -322,84 +291,32 @@ WireLib.ThrusterEffectDraw.fire_smoke_big = function(self)
 	util.Effect( "ThumperDust ", effectdata )
 end
 
-WireLib.ThrusterEffectThink.smoke = function(self)
-
-	self.SmokeTimer = self.SmokeTimer or 0
-	if ( self.SmokeTimer > CurTime() ) then return end
-
-	self.SmokeTimer = CurTime() + 0.015
-
-	local vOffset = self:LocalToWorld(self:GetOffset()) + Vector( math.Rand( -3, 3 ), math.Rand( -3, 3 ), math.Rand( -3, 3 ) )
-	local vNormal = self:CalcNormal()
-
-		local particle = emitter:Add( "particles/smokey", vOffset )
-			particle:SetVelocity( vNormal * math.Rand( 10, 30 ) )
-			particle:SetDieTime( 2.0 )
-			particle:SetStartAlpha( math.Rand( 50, 150 ) )
-			particle:SetStartSize( math.Rand( 16, 32 ) )
-			particle:SetEndSize( math.Rand( 64, 128 ) )
-			particle:SetRoll( math.Rand( -0.2, 0.2 ) )
-			particle:SetColor( 200, 200, 210 )
-end
-
-WireLib.ThrusterEffectThink.smoke_firecolors = function(self)
-
-	self.SmokeTimer = self.SmokeTimer or 0
-	if ( self.SmokeTimer > CurTime() ) then return end
-
-	self.SmokeTimer = CurTime() + 0.015
-
-	local vOffset = self:LocalToWorld(self:GetOffset()) + Vector( math.Rand( -3, 3 ), math.Rand( -3, 3 ), math.Rand( -3, 3 ) )
-	local vNormal = self:CalcNormal()
-
-		local particle = emitter:Add( "particles/smokey", vOffset )
-			particle:SetVelocity( vNormal * math.Rand( 10, 30 ) )
-			particle:SetDieTime( 2.0 )
-			particle:SetStartAlpha( math.Rand( 50, 150 ) )
-			particle:SetStartSize( math.Rand( 16, 32 ) )
-			particle:SetEndSize( math.Rand( 64, 128 ) )
-			particle:SetRoll( math.Rand( -0.2, 0.2 ) )
-			particle:SetColor(math.random(220,255),math.random(110,220),0 )
-end
-
-WireLib.ThrusterEffectThink.smoke_random = function(self)
+local function smoke(color)
+	return function(self)
 
-	self.SmokeTimer = self.SmokeTimer or 0
-	if ( self.SmokeTimer > CurTime() ) then return end
+		self.SmokeTimer = self.SmokeTimer or 0
+		if ( self.SmokeTimer > CurTime() ) then return end
 
-	self.SmokeTimer = CurTime() + 0.015
+		self.SmokeTimer = CurTime() + 0.015
 
-	local vOffset = self:LocalToWorld(self:GetOffset()) + Vector( math.Rand( -3, 3 ), math.Rand( -3, 3 ), math.Rand( -3, 3 ) )
-	local vNormal = self:CalcNormal()
+		local vOffset = self:LocalToWorld(self:GetOffset()) + Vector( math.Rand( -3, 3 ), math.Rand( -3, 3 ), math.Rand( -3, 3 ) )
+		local vNormal = self:CalcNormal()
 
-		local particle = emitter:Add( "particles/smokey", vOffset )
-			particle:SetVelocity( vNormal * math.Rand( 10, 30 ) )
-			particle:SetDieTime( 2.0 )
-			particle:SetStartAlpha( math.Rand( 50, 150 ) )
-			particle:SetStartSize( math.Rand( 16, 32 ) )
-			particle:SetEndSize( math.Rand( 64, 128 ) )
-			particle:SetRoll( math.Rand( -0.2, 0.2 ) )
-			particle:SetColor( math.random(100,255),math.random(100,255),math.random(100,255) )
+			local particle = emitter:Add( "particles/smokey", vOffset )
```

Comment:
This is one extra tab over from the rest of the scope. Few other places where this is the case as well. Looks like it's just the result of how the old code was formatted, should this be fixed here or for later clean up?

Reply:
Cleaned up!